source/rust_verify_test/tests/raw_ptrs.rs

```diff
@@ -0,0 +1,170 @@
+#![feature(rustc_private)]
+#[macro_use]
+mod common;
+use common::*;
+
+test_verify_one_file! {
+    #[test] basics verus_code! {
+        use vstd::prelude::*;
+        use vstd::raw_ptr::*;
+
+        fn test_mut_const_casts(x: *mut u8) {
+            let y = x as *const u8;
+            let z = y as *const u8;
+            assert(z == x);
+            assert(z == y);
+        }
+
+        fn test_addr_doesnt_imply_eq(x: *mut u8, y: *mut u8) {
+            assume(x@.addr == y@.addr);
+            assert(x == y); // FAILS
+        }
```

Comment:
If I understand correctly, the following test should fail as well (and it does):

```
fn test(x: *const u8, y: *const u8) {
    if x == y {
        assert(x == y);
    }
}
```

Could this be added to the tests?